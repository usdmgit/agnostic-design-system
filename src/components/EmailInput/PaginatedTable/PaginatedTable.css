@import '../../styles/theme/index.css';

.paginated-footer .paginated-table-action-btns {
  align-items: var(--paginated-table-align-items, center);
  display: var(--paginated-table-display, flex);
  justify-content: var(--paginated-table-justify-content, flex-end);
  overflow: var(--paginated-table-overflow, hidden);
}

.paginated-table-action-btns .btn-table-paginated {
  font-family: var(--paginated-table-font-family, 'Inter');
  font-size: var(--paginated-table-font-size, 14px);
  font-weight: var(--paginated-table-font-weight, 300);
  padding: var(--paginated-table-padding, 10px);
}

.number-of-pages {
  text-align: var(--paginated-table-page-number-text-align, end);
  white-space: var(--paginated-table-page-number-white-space, normal);
}

.number-of-pages:after {
  color: var(--paginated-table-separator-color, var(--structure-color-800));
  content: '|';
  margin: var(--paginated-table-separator-margin, 0 0 0 20px);
}

.number-of-pages-text {
  font-size: var(--paginated-table-page-number-text, 14px);
}

.number-of-pages-input {
  max-width: var(--paginated-table-page-input-max-width, 35px);
  display: inline-table;
  margin-right: 10px;
  --input-padding: 5px 10px;
  text-align: center;
}

.pagination-items {
  margin: var(--paginated-table-total-number-items-margin, 0);
}

.pagination-items,
.number-of-pages-text {
  font-size: var(--paginated-table-footer-font-size, 14px);
}

.pagination-items,
.number-of-pages,
.paginated-table-action-btns {
  vertical-align: baseline;
}

.paginated-table-pagination {
  margin: 0;
  padding: var(--paginated-table-pagination-padding, 0);
}

.paginated-table-pages {
  display: flex;
  align-items: center;
}

.paginated-footer-wrapper {
  display: var(--paginated-table-footer-wrapper-display, flex);
  flex-direction: var(--paginated-table-footer-wrapper-flex-direction, row);
  justify-content: space-between;
  align-items: center;
}
