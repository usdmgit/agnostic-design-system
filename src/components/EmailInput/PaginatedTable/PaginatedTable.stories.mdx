# PaginatedTable Component

## Type: Composite

The PaginatedTable contains the following components: [Table](https://design-system.codelitt.dev/?path=/docs/components-table--default), [Pagination](https://design-system.codelitt.dev/?path=/docs/components-pagination--default) and [NumericInput](https://design-system.codelitt.dev/?path=/docs/components-numeric-input--default).

## PaginatedTable Props

|        Prop        |              Type               |                                                  Description                                                   |        Default        | Mandatory |
| :----------------: | :-----------------------------: | :------------------------------------------------------------------------------------------------------------: | :-------------------: | :-------: |
|   buttonNextIcon   |         React.ReactNode         |                               Displays the icon for the next pagination button.                                | <span> &rarr; </span> |   false   |
|   buttonPrevIcon   |         React.ReactNode         |                             Displays the icon for the previous pagination button.                              | <span> &larr; </span> |   false   |
|    currentPage     |             number              |                               Specifies the current page visible for the table.                                |           -           |   true    |
|  handlePageChange  |            function             |               Specifies what should happens when a change event is triggered on the page number.               |           -           |   true    |
|     onRowClick     |            function             |     Specifies what happens when a user clicks in a specific row. Receives the row index and the row data.      |           -           |   false   |
|  onRowMouseEnter   |            function             |     Specifies what happens when a user hovers in a specific row. Receives the row index and the row data.      |           -           |   false   |
|  onRowMouseLeave   |            function             | Specifies what happens when a user ceases hovering in a specific row. Receives the row index and the row data. |           -           |   false   |
|     headerList     | array of strings or react nodes |                      Specifies the list to be used to populate the PaginatedTable header.                      |           -           |   true    |
| itemsOnCurrentPage | array of strings or react nodes |                           Specifies what should be the items displayed on the table.                           |           -           |   true    |
|       limit        |             number              |                          Specifies the maximum number of items each page should have.                          |           -           |   true    |
|  renderEmptyState  |      () => React.ReactNode      |                  Render an ReactNode in case there are no items in the table to be displayed.                  |           -           |   false   |
| totalNumberOfItems |             number              |                        Defines the total number of items to be displayed on the table.                         |           -           |   true    |
| variablesClassName |             string              |                       Specifies the custom class to apply styles to the PaginatedTable.                        |           -           |   false   |

## Getting Started

```javascript
import { PaginatedTable } from '@codelittinc/agnostic-design-system';

const props = {
  currentPage: 1,
  handlePageChange: e => e,
  headerList: ['First Name', 'Last Name', 'Handle', 'User Type', 'Status'],
  itemsOnCurrentPage: [
    ['John', 'Oto', '@timo', 'type1', 'Active'],
    ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
    ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
    ['Lary', 'King', '@timo', 'type1', 'Inactive'],
    ['Helen', 'Bonham', '@timo', 'type2', 'Active'],
    ['Felix', 'Dougherty', '@timo', 'type1', 'Active'],
    ['Joseph', 'Thomas', '@timo', 'type1', 'Inactive'],
    ['Richard', 'Cardenas', '@timo', 'type1', 'Active']
  ],
  totalNumberOfItems: 15
};

const PaginatedTable = props => {
  return <PaginatedTable {...props} />;
};

export default PaginatedTable;
```

## CSS Variables Available

Given the nature of a composite component, in addition to its own, the PaginatedTable utilizes CSS variables from the Table, Pagination and NumericInput components.
PaginatedTable variables are listed in the table below.

### Variables of the PaginatedTable component

|      Property      |    Attribute    | State |
| :----------------: | :-------------: | :---: |
|                    |   align-items   |       |
|                    |   font-family   |       |
|                    |    font-size    |       |
|                    |   font-weight   |       |
|                    | justify-content |       |
|                    |     padding     |       |
|                    |    overflow     |       |
|       button       |   font-family   |       |
|       button       |    font-size    |       |
|       button       |   font-weight   |       |
|       button       |     padding     |       |
|      buttons       |     display     |       |
|   clickable-row    |     cursor      |       |
|       footer       |    font-size    |       |
|   footer-wrapper   |     display     |       |
|   footer-wrapper   | flex-direction  |       |
|     page-input     |    max-width    |       |
|    page-number     |   text-align    |       |
|    page-number     |   white-space   |       |
|     separator      |      color      |       |
|     separator      |     margin      |       |
| total-number-items |     margin      |       |
