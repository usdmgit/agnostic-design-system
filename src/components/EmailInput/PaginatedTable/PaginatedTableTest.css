@import '../../styles/theme/index.css';

.custom-table {
  --button-background-color: red;
  --button-color: black;
  --button-width: 69px;
  --button-height: 32px;
  --paginated-table-footer-wrapper-flex-direction: column;
  --paginated-table-page-number-text-align: normal;
  --paginated-table-footer-font-size: 12px;
  --paginated-table-total-number-items-margin: 0 0 10px 0;
  --table-padding: 15px;
  --table-cell-padding: 10px;
  --table-footer-padding: 20px;
}
