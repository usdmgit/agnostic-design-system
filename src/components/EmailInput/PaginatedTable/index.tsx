import React, { useState } from 'react';
import NumericInput from '@/components/NumericInput';
import Pagination from '@/components/Pagination';
import Table from '@/components/Table';
import classNames from 'classnames';
import ArrowLeft from '@/assets/images/icons/web/arrow-left.svg';
import ArrowRight from '@/assets/images/icons/web/arrow-right.svg';
import { isEmpty } from 'lodash';
import styles from '@/components/PaginatedTable/PaginatedTable.css';

export interface Props {
  currentPage: number;
  handlePageChange: (value: number) => void;
  headerList: React.ReactNode[];
  id?: string;
  itemsOnCurrentPage: React.ReactNode[][];
  limit: number;
  onRowClick?: (row: React.ReactNode, rowIndex: number) => void;
  onRowMouseEnter?: (row: React.ReactNode, rowIndex: number) => void;
  onRowMouseLeave?: (row?: React.ReactNode, rowIndex?: number) => void;
  renderEmptyState?: () => React.ReactNode;
  totalNumberOfItems: number;
  variablesClassName?: string;
}

const PaginatedTable = (props: Props) => {
  const {
    currentPage,
    handlePageChange,
    headerList,
    id,
    itemsOnCurrentPage,
    limit,
    onRowClick,
    onRowMouseEnter,
    onRowMouseLeave,
    renderEmptyState,
    totalNumberOfItems,
    variablesClassName
  } = props;

  const [uniqueId] = useState(id || Math.floor(Math.random() * 1000 + 1));

  const numberOfPages = Math.ceil(totalNumberOfItems / limit);

  const handleChange = event => {
    const { value } = event.target;

    if (value > 0 || isEmpty(value)) {
      handlePageChange(value);
    }
  };

  const itemsRangeBegin = () => {
    if (itemsOnCurrentPage) {
      return currentPage > 0 ? (currentPage - 1) * limit + 1 : 1;
    }
    return 0;
  };

  const itemsRangeEnd = () => {
    if (itemsOnCurrentPage) {
      const begin = itemsRangeBegin();
      const total = begin + itemsOnCurrentPage.length - 1;
      return Math.min(total, totalNumberOfItems);
    }
    return 0;
  };

  const widthFirstFooterCell = headerList.length;

  return (
    <>
      {!totalNumberOfItems && renderEmptyState ? (
        renderEmptyState()
      ) : (
        <Table
          headerList={headerList}
          tableContent={itemsOnCurrentPage}
          variablesClassName={variablesClassName}
          onRowClick={onRowClick}
          onRowMouseEnter={onRowMouseEnter}
          onRowMouseLeave={onRowMouseLeave}
          renderFooter={() => (
            <tfoot className={styles['paginated-footer']}>
              <tr>
                <td colSpan={widthFirstFooterCell}>
                  <div className={styles['paginated-footer-wrapper']}>
                    <div className={styles['pagination-items']}>
                      {`${itemsRangeBegin()} - ${itemsRangeEnd()} of ${totalNumberOfItems} items`}
                    </div>
                    <div className={styles['paginated-table-pages']}>
                      <div className={styles['number-of-pages']}>
                        <NumericInput
                          id={`paginated-table-input-${uniqueId}`}
                          max={numberOfPages}
                          onChange={handleChange}
                          onStateChange={() => {}}
                          onFocus={() => {}}
                          scale={0}
                          size='medium'
                          value={currentPage.toString()}
                          variablesClassName={styles['number-of-pages-input']}
                          positive
                        />
                        <span className={styles['number-of-pages-text']}>
                          of {numberOfPages} pages
                        </span>
                      </div>
                      <div className={styles['paginated-table-action-btns']}>
                        <Pagination
                          totalPages={numberOfPages}
                          numberOfPagesToShow={0}
                          currentPage={currentPage}
                          onSelectPage={handlePageChange}
                          buttonNextLabel='Next'
                          buttonNextIcon={<ArrowRight />}
                          buttonNextIconPosition='right'
                          buttonPrevLabel='Prev'
                          buttonPrevIcon={<ArrowLeft />}
                          buttonPrevIconPosition='left'
                          buttonVariablesClassName={classNames(
                            styles['btn-table-paginated'],
                            variablesClassName
                          )}
                          variablesClassName={classNames(
                            styles['paginated-table-pagination'],
                            variablesClassName
                          )}
                        />
                      </div>
                    </div>
                  </div>
                </td>
              </tr>
            </tfoot>
          )}
        />
      )}
    </>
  );
};

export default PaginatedTable;

PaginatedTable.defaultProps = {
  headerList: [],
  tableContent: [[]]
};
