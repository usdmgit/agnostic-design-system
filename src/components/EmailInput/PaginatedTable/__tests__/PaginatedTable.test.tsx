import React, { useState } from 'react';
// eslint-disable-next-line no-unused-vars
import PaginatedTable, { Props } from '@/components/PaginatedTable';
import { mount } from 'enzyme';
import { render, screen, fireEvent, within } from '@testing-library/react';
import '@testing-library/jest-dom';

describe('PaginatedTable', () => {
  it('renders correctly', () => {
    const wrapper = mount(
      <PaginatedTable
        currentPage={1}
        handlePageChange={() => {}}
        headerList={['First Name', 'Last Name', 'Handle', 'User Type', 'Status']}
        id='paginated-table'
        itemsOnCurrentPage={[
          ['John', 'Oto', '@timo', 'type1', 'Active'],
          ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
          ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
          ['Lary', 'King', '@timo', 'type1', 'Inactive'],
          ['Helen', 'Bonham', '@timo', 'type2', 'Active']
        ]}
        limit={5}
        totalNumberOfItems={15}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('Paginated input', () => {
    it('accepts the number of the page to display', () => {
      renderPaginatedTable();
      const PaginatedTableList = screen.getByTestId('table-element');
      const paginationInput = PaginatedTableList.querySelector('input')!;
      expect(paginationInput).toHaveValue('1');
      fireEvent.change(paginationInput, { target: { value: '3' } });
      expect(paginationInput).toHaveValue('3');
    });

    it('does not accept numbers smaller than 1', () => {
      renderPaginatedTable();
      const PaginatedTableList = screen.getByTestId('table-element');
      const paginationInput = PaginatedTableList.querySelector('input')!;
      expect(paginationInput).toHaveValue('1');
    });

    it('does not accept numbers bigger than the total number of pages', () => {
      renderPaginatedTable();
      const PaginatedTableList = screen.getByTestId('table-element');
      const paginationInput = PaginatedTableList.querySelector('input')!;
      expect(paginationInput).toHaveValue('1');
      fireEvent.change(paginationInput, { target: { value: '5' } });
      expect(paginationInput).toHaveValue('1');
    });
  });

  describe('Number of items displayed', () => {
    it('changes if you change the page', () => {
      renderPaginatedTable({
        itemsOnCurrentPage: [
          ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
          ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
          ['Lary', 'King', '@timo', 'type1', 'Inactive'],
          ['Helen', 'Bonham', '@timo', 'type2', 'Active'],
          ['Felix', 'Dougherty', '@timo', 'type1', 'Active'],
          ['Joseph', 'Thomas', '@timo', 'type1', 'Inactive'],
          ['Richard', 'Cardenas', '@timo', 'type1', 'Active'],
          ['Karl', 'Roy', '@twitter', 'type2', 'Active'],
          ['Sean', 'Steppe', '@timo', 'type1', 'Inactive'],
          ['Samuel', 'Marriott', '@twitter', 'type1', 'Inactive']
        ],
        limit: 10
      });
      const PaginatedTableItems = screen
        .getByTestId('table-element')
        .querySelector('.pagination-items')!;

      const paginationInput = screen.getByTestId('table-element').querySelector('input')!;

      expect(PaginatedTableItems).toHaveTextContent(/1 - 10 of 15 items/);
      fireEvent.change(paginationInput, { target: { value: 2 } });
      expect(PaginatedTableItems).toHaveTextContent(/11 - 15 of 15 items/);
    });

    it('shows the right number of items per page', () => {
      renderPaginatedTable({
        itemsOnCurrentPage: [
          ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
          ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
          ['Lary', 'King', '@timo', 'type1', 'Inactive'],
          ['Helen', 'Bonham', '@timo', 'type2', 'Active'],
          ['Felix', 'Dougherty', '@timo', 'type1', 'Active'],
          ['Joseph', 'Thomas', '@timo', 'type1', 'Inactive'],
          ['Richard', 'Cardenas', '@timo', 'type1', 'Active']
        ],
        limit: 7,
        totalNumberOfItems: 10
      });
      const PaginatedTableItems = screen
        .getByTestId('table-element')
        .querySelector('.pagination-items')!;

      const paginationInput = screen.getByTestId('table-element').querySelector('input')!;

      expect(PaginatedTableItems).toHaveTextContent(/1 - 7 of 10 items/);
      fireEvent.change(paginationInput, { target: { value: 2 } });
      expect(PaginatedTableItems).toHaveTextContent(/8 - 10 of 10 items/);
    });
  });

  describe('Usage of the prev button', () => {
    it('is disabled if you are on the first page', () => {
      renderPaginatedTable();
      const table = screen.getByTestId('table-element');
      const prevButton = within(table).getByText(/Prev/i).closest('button') as HTMLButtonElement;
      expect(prevButton.disabled).toBeTruthy();
    });

    it('is enabled if previous pages exist', () => {
      renderPaginatedTable({ currentPage: 2 });
      const table = screen.getByTestId('table-element');
      const prevButton = within(table).getByText(/Prev/i).closest('button') as HTMLButtonElement;
      expect(prevButton.disabled).not.toBeTruthy();
    });

    it('calls handlePageChange with the correct page number', () => {
      const handlePageChange = jest.fn();
      renderPaginatedTable({ handlePageChange, currentPage: 2 });
      fireEvent.click(screen.getByText(/Prev/));
      expect(handlePageChange).toHaveBeenCalledWith(1);
    });
  });

  describe('Usage of the next button', () => {
    it('is disabled if you are on the last page', () => {
      renderPaginatedTable({ currentPage: 3 });
      const table = screen.getByTestId('table-element');
      const nextButton = within(table).getByText(/Next/i).closest('button') as HTMLButtonElement;
      expect(nextButton.disabled).toBeTruthy();
    });

    it('is enabled if next pages exist', () => {
      renderPaginatedTable({ currentPage: 2 });
      const table = screen.getByTestId('table-element');
      const nextButton = within(table).getByText(/Next/i).closest('button') as HTMLButtonElement;
      expect(nextButton.disabled).not.toBeTruthy();
    });

    it('calls handlePageChange with the correct page number', () => {
      const handlePageChange = jest.fn();
      renderPaginatedTable({ handlePageChange, currentPage: 2 });
      fireEvent.click(screen.getByText(/Next/));
      expect(handlePageChange).toHaveBeenCalledWith(3);
    });
  });
});

function Wrapper(props) {
  const [page, setPage] = useState(1);

  return (
    <PaginatedTable
      currentPage={page}
      id='paginated-table'
      itemsOnCurrentPage={[
        ['John', 'Oto', '@timo', 'type1', 'Active'],
        ['Mark', 'Howard', '@mdo', 'type2', 'Inactive'],
        ['Jacob', 'Martin', '@mdo', 'type1', 'Active'],
        ['Lary', 'King', '@timo', 'type1', 'Inactive'],
        ['Helen', 'Bonham', '@timo', 'type2', 'Active']
      ]}
      handlePageChange={setPage}
      headerList={['First Name', 'Last Name', 'Handle', 'User Type', 'Status']}
      totalNumberOfItems={15}
      limit={5}
      {...props}
    />
  );
}

function renderPaginatedTable(props: Partial<Props> = {}) {
  return render(<Wrapper {...props} />);
}
