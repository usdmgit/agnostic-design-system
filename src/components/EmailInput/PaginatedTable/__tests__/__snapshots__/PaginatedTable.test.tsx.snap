// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginatedTable renders correctly 1`] = `
<PaginatedTable
  currentPage={1}
  handlePageChange={[Function]}
  headerList={
    Array [
      "First Name",
      "Last Name",
      "Handle",
      "User Type",
      "Status",
    ]
  }
  id="paginated-table"
  itemsOnCurrentPage={
    Array [
      Array [
        "John",
        "Oto",
        "@timo",
        "type1",
        "Active",
      ],
      Array [
        "Mark",
        "Howard",
        "@mdo",
        "type2",
        "Inactive",
      ],
      Array [
        "Jacob",
        "Martin",
        "@mdo",
        "type1",
        "Active",
      ],
      Array [
        "Lary",
        "King",
        "@timo",
        "type1",
        "Inactive",
      ],
      Array [
        "Helen",
        "Bonham",
        "@timo",
        "type2",
        "Active",
      ],
    ]
  }
  limit={5}
  tableContent={
    Array [
      Array [],
    ]
  }
  totalNumberOfItems={15}
>
  <Table
    headerList={
      Array [
        "First Name",
        "Last Name",
        "Handle",
        "User Type",
        "Status",
      ]
    }
    renderFooter={[Function]}
    tableContent={
      Array [
        Array [
          "John",
          "Oto",
          "@timo",
          "type1",
          "Active",
        ],
        Array [
          "Mark",
          "Howard",
          "@mdo",
          "type2",
          "Inactive",
        ],
        Array [
          "Jacob",
          "Martin",
          "@mdo",
          "type1",
          "Active",
        ],
        Array [
          "Lary",
          "King",
          "@timo",
          "type1",
          "Inactive",
        ],
        Array [
          "Helen",
          "Bonham",
          "@timo",
          "type2",
          "Active",
        ],
      ]
    }
  >
    <table
      className="table-container"
      data-testid="table-element"
    >
      <thead
        data-testid="table-header"
      >
        <tr>
          <th
            key="0"
          >
            First Name
          </th>
          <th
            key="1"
          >
            Last Name
          </th>
          <th
            key="2"
          >
            Handle
          </th>
          <th
            key="3"
          >
            User Type
          </th>
          <th
            key="4"
          >
            Status
          </th>
        </tr>
      </thead>
      <tbody
        data-testid="table-body"
      >
        <tr
          className="non-clickable-row"
          key="0"
          onClick={[Function]}
          onKeyPress={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
        >
          <td
            key="0 - 0"
          >
             
            John
          </td>
          <td
            key="0 - 1"
          >
             
            Oto
          </td>
          <td
            key="0 - 2"
          >
             
            @timo
          </td>
          <td
            key="0 - 3"
          >
             
            type1
          </td>
          <td
            key="0 - 4"
          >
             
            Active
          </td>
        </tr>
        <tr
          className="non-clickable-row"
          key="1"
          onClick={[Function]}
          onKeyPress={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
        >
          <td
            key="1 - 0"
          >
             
            Mark
          </td>
          <td
            key="1 - 1"
          >
             
            Howard
          </td>
          <td
            key="1 - 2"
          >
             
            @mdo
          </td>
          <td
            key="1 - 3"
          >
             
            type2
          </td>
          <td
            key="1 - 4"
          >
             
            Inactive
          </td>
        </tr>
        <tr
          className="non-clickable-row"
          key="2"
          onClick={[Function]}
          onKeyPress={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
        >
          <td
            key="2 - 0"
          >
             
            Jacob
          </td>
          <td
            key="2 - 1"
          >
             
            Martin
          </td>
          <td
            key="2 - 2"
          >
             
            @mdo
          </td>
          <td
            key="2 - 3"
          >
             
            type1
          </td>
          <td
            key="2 - 4"
          >
             
            Active
          </td>
        </tr>
        <tr
          className="non-clickable-row"
          key="3"
          onClick={[Function]}
          onKeyPress={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
        >
          <td
            key="3 - 0"
          >
             
            Lary
          </td>
          <td
            key="3 - 1"
          >
             
            King
          </td>
          <td
            key="3 - 2"
          >
             
            @timo
          </td>
          <td
            key="3 - 3"
          >
             
            type1
          </td>
          <td
            key="3 - 4"
          >
             
            Inactive
          </td>
        </tr>
        <tr
          className="non-clickable-row"
          key="4"
          onClick={[Function]}
          onKeyPress={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          tabIndex={0}
        >
          <td
            key="4 - 0"
          >
             
            Helen
          </td>
          <td
            key="4 - 1"
          >
             
            Bonham
          </td>
          <td
            key="4 - 2"
          >
             
            @timo
          </td>
          <td
            key="4 - 3"
          >
             
            type2
          </td>
          <td
            key="4 - 4"
          >
             
            Active
          </td>
        </tr>
      </tbody>
      <tfoot
        className="paginated-footer"
      >
        <tr>
          <td
            colSpan={5}
          >
            <div
              className="paginated-footer-wrapper"
            >
              <div
                className="pagination-items"
              >
                1 - 5 of 15 items
              </div>
              <div
                className="paginated-table-pages"
              >
                <div
                  className="number-of-pages"
                >
                  <ForwardRef
                    id="paginated-table-input-paginated-table"
                    max={3}
                    onChange={[Function]}
                    onFocus={[Function]}
                    onStateChange={[Function]}
                    positive={true}
                    scale={0}
                    size="medium"
                    value="1"
                    variablesClassName="number-of-pages-input"
                  >
                    <ForwardRef
                      filters={
                        Array [
                          Object {
                            "test": /\\[\\^\\\\d\\.,-\\]/,
                            "type": "RegExp",
                          },
                          Object {
                            "test": [Function],
                            "type": "function",
                          },
                          Object {
                            "test": [Function],
                            "type": "function",
                          },
                          Object {
                            "test": [Function],
                            "type": "function",
                          },
                        ]
                      }
                      id="paginated-table-input-paginated-table"
                      onChange={[Function]}
                      onFocus={[Function]}
                      onStateChange={[Function]}
                      size="medium"
                      validations={Array []}
                      value="1"
                      variablesClassName="number-of-pages-input"
                    >
                      <div
                        className="number-of-pages-input container"
                      >
                        <div
                          className="input--field-container"
                        >
                          <div
                            className="input--container"
                          >
                            <input
                              className="input input--medium"
                              id="paginated-table-input-paginated-table"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              value="1"
                            />
                          </div>
                        </div>
                      </div>
                    </ForwardRef>
                  </ForwardRef>
                  <span
                    className="number-of-pages-text"
                  >
                    of 
                    3
                     pages
                  </span>
                </div>
                <div
                  className="paginated-table-action-btns"
                >
                  <Pagination
                    buttonNextIcon={<SvgArrowRight />}
                    buttonNextIconPosition="right"
                    buttonNextLabel="Next"
                    buttonPrevIcon={<SvgArrowLeft />}
                    buttonPrevIconPosition="left"
                    buttonPrevLabel="Prev"
                    buttonVariablesClassName="btn-table-paginated"
                    currentPage={1}
                    numberOfPagesToShow={0}
                    onSelectPage={[Function]}
                    totalPages={3}
                    variablesClassName="paginated-table-pagination"
                  >
                    <ul
                      className="pagination-list paginated-table-pagination"
                      data-testid="pagination-element"
                    >
                      <li>
                        <ForwardRef
                          appendIcon={<SvgArrowLeft />}
                          disabled={true}
                          iconPosition="left"
                          onClick={[Function]}
                          size="medium"
                          text="Prev"
                          variablesClassName="btn-table-paginated"
                          withAppendIcon={true}
                        >
                          <ForwardRef
                            appendIcon={<SvgArrowLeft />}
                            disabled={true}
                            iconPosition="left"
                            onClick={[Function]}
                            size="medium"
                            text="Prev"
                            variablesClassName="btn-table-paginated"
                            withAppendIcon={true}
                          >
                            <ForwardRef
                              appendIcon={<SvgArrowLeft />}
                              className="button button--neutral"
                              disabled={true}
                              fixed={false}
                              iconPosition="left"
                              onClick={[Function]}
                              size="medium"
                              text="Prev"
                              variablesClassName="btn-table-paginated"
                              withAppendIcon={true}
                            >
                              <button
                                className="button button--neutral button button--medium btn-table-paginated"
                                disabled={true}
                                onClick={[Function]}
                                type="button"
                              >
                                <div
                                  className="append-icon-left"
                                >
                                  <div
                                    className="button--append-icon"
                                  >
                                    <SvgArrowLeft>
                                      <svg
                                        data-file-name="SvgArrowLeft"
                                      />
                                    </SvgArrowLeft>
                                  </div>
                                </div>
                                <span
                                  className="button-text"
                                >
                                  Prev
                                </span>
                              </button>
                            </ForwardRef>
                          </ForwardRef>
                        </ForwardRef>
                      </li>
                      <li>
                        <ForwardRef
                          appendIcon={<SvgArrowRight />}
                          disabled={false}
                          iconPosition="right"
                          onClick={[Function]}
                          size="medium"
                          text="Next"
                          variablesClassName="btn-table-paginated"
                          withAppendIcon={true}
                        >
                          <ForwardRef
                            appendIcon={<SvgArrowRight />}
                            disabled={false}
                            iconPosition="right"
                            onClick={[Function]}
                            size="medium"
                            text="Next"
                            variablesClassName="btn-table-paginated"
                            withAppendIcon={true}
                          >
                            <ForwardRef
                              appendIcon={<SvgArrowRight />}
                              className="button button--neutral"
                              disabled={false}
                              fixed={false}
                              iconPosition="right"
                              onClick={[Function]}
                              size="medium"
                              text="Next"
                              variablesClassName="btn-table-paginated"
                              withAppendIcon={true}
                            >
                              <button
                                className="button button--neutral button button--medium btn-table-paginated"
                                disabled={false}
                                onClick={[Function]}
                                type="button"
                              >
                                <span
                                  className="button-text"
                                >
                                  Next
                                </span>
                                <div
                                  className="append-icon-right"
                                >
                                  <div
                                    className="button--append-icon"
                                  >
                                    <SvgArrowRight>
                                      <svg
                                        data-file-name="SvgArrowRight"
                                      />
                                    </SvgArrowRight>
                                  </div>
                                </div>
                              </button>
                            </ForwardRef>
                          </ForwardRef>
                        </ForwardRef>
                      </li>
                    </ul>
                  </Pagination>
                </div>
              </div>
            </div>
          </td>
        </tr>
      </tfoot>
    </table>
  </Table>
</PaginatedTable>
`;
